### PR TITLE
#704: Removing ArtifactResolver

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Second stage: Removing ArtifactResolver.

@slawekjaranowski please review

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
Also managed to reduce the `settings` property (it's already available via `MavenSession`).

File: versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java [hunk 1]
```diff
@@ -479,13 +464,21 @@ private List<IgnoreVersion> getIgnoredVersions( Artifact artifact )
...
-                        : mavenSession.getCurrentProject().getRemoteArtifactRepositories(),
-                localRepository );
+        try
+        {
+            aetherRepositorySystem.resolveArtifact( mavenSession.getRepositorySession(),
+                    new ArtifactRequest( toArtifact( artifact ),
+                            usePluginRepositories
+                                    ? mavenSession.getCurrentProject().getRemotePluginRepositories()
+                                    : mavenSession.getCurrentProject().getRemoteProjectRepositories(),
+                            getClass().getName() ) );
```
Comment (slawekjaranowski) [inline versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java:476]: We need process response and update source artifact, like:
```
rtifactResult artifactResult  = resolveArtifact( ... );

        artifact.setFile( artifactResult.getArtifact().getFile() );
        artifact.setVersion( artifactResult.getArtifact().getVersion() );
        artifact.setResolved( true );
```

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java (hunk 1 above)
Comment (ajarmoniuk) [reply versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java]: Done

**Comment by ajarmoniuk** [comment]
Changes applied.

**Comment by ajarmoniuk** [comment]
@slawekjaranowski rebased and resolved conflicts

File: versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java [hunk 2]
```diff
@@ -479,13 +465,30 @@ private List<IgnoreVersion> getIgnoredVersions( Artifact artifact )
...
+            artifact.setFile( artifactResult.getArtifact().getFile() );
+            artifact.setVersion( artifactResult.getArtifact().getVersion() );
+            artifact.setResolved( artifactResult.isResolved() );
+            if ( !artifactResult.getExceptions().isEmpty() )
+            {
+                getLog().warn( "Exceptions thrown while trying to resolve artifact " + artifact );
+                artifactResult.getExceptions().forEach( e ->
+                        getLog().warn( "\t" + e.getMessage() ) );
+            }
```
Comment (slawekjaranowski) [inline versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java:486]: I think that never occurs in this place. 

When exceptions list is set also exception is thrown.

Comment (ajarmoniuk) [reply versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java]: It looks like you're right. Removed.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**